libnavui-app/src/main/java/com/mapbox/navigation/ui/app/internal/SharedApp.kt

```diff
@@ -40,24 +44,31 @@ object SharedApp {
     val routeStateController = RouteStateController(store)
     val routePreviewStateController = RoutePreviewStateController(store)
     private val navigationObservers: Array<MapboxNavigationObserver> = arrayOf(
-        destinationStateController,
-        tripSessionStarterStateController,
-        audioGuidanceStateController,
-        locationStateController,
         routeStateController,
-        routePreviewStateController,
         cameraStateController,
+        locationStateController,
         navigationStateController,
+        destinationStateController,
+        routePreviewStateController,
+        audioGuidanceStateController,
+        tripSessionStarterStateController,
     )
 
     fun setup(
         context: Context,
-        audioGuidance: MapboxAudioGuidance? = null
+        audioGuidance: MapboxAudioGuidance? = null,
+        routeAlternativeContract: Provider<RouteAlternativeContract>? = null
```

Comment:
Is `Provider` really necessary? It is like wrapping a lambda with another lambda.